wandb/apis/workspaces/expr.py

```diff
@@ -0,0 +1,308 @@
+            del self._backward[self._forward[key]]
+        if value in self._backward:
+            raise ValueError(f"Duplicate value found: {value}")
+        self._forward[key] = value
+        self._backward[value] = key
+
+    def __delitem__(self, key):
+        value = self._forward.pop(key)
+        del self._backward[value]
+
+    def __iter__(self):
+        return iter(self._forward)
+
+    def __len__(self):
+        return len(self._forward)
+
+    def __repr__(self):
+        return repr(self._forward)
+
+    def __contains__(self, key):
+        return key in self._forward
+
+    @property
+    def inv(self):
+        return self._backward
+
+
+FE_BE_NAME_MAP = InvertableDict(
+    {
+        "ID": "name",
+        "Name": "displayName",
+        "Tags": "tags",
+        "State": "state",
+        "CreatedTimestamp": "createdAt",
+        "Runtime": "duration",
+        "User": "username",
+        "Sweep": "sweep",
+        "Group": "group",
+        "JobType": "jobType",
+        "Hostname": "host",
+        "UsingArtifact": "inputArtifacts",
+        "OutputtingArtifact": "outputArtifacts",
+        "Step": "_step",
+        "RelativeTime(Wall)": "_absolute_runtime",
+        "RelativeTime(Process)": "_runtime",
+        "WallTime": "_timestamp",
+    }
+)
+
+
+# Mapping custom operators to Python operators
+OPERATOR_MAP = InvertableDict(
+    {
+        "AND": "and",
+        "OR": "or",
+        "=": "==",
+        "!=": "!=",
+        "<": "<",
+        "<=": "<=",
+        ">": ">",
+        ">=": ">=",
+        "IN": "in",
+        "NIN": "not in",
+    }
+)
+
+SECTION_MAP = InvertableDict(
+    {
+        "Metric": "run",
+        "Summary": "summary",
+        "Config": "config",
+        "Tags": "tags",
+        "KeysInfo": "keys_info",
+    }
+)
+
+
+@dataclass(eq=False, frozen=True)
+class BaseMetric:
+    name: str
+    section: ClassVar[str]  # declared in subclasses
+
+    def __eq__(self, other: Any) -> "FilterExpr":
+        return FilterExpr.create("=", self, other)
+
+    def __ne__(self, other: Any) -> "FilterExpr":
+        return FilterExpr.create("!=", self, other)
+
+    def __lt__(self, other: Any) -> "FilterExpr":
+        return FilterExpr.create("<", self, other)
+
+    def __le__(self, other: Any) -> "FilterExpr":
+        return FilterExpr.create("<=", self, other)
+
+    def __gt__(self, other: Any) -> "FilterExpr":
+        return FilterExpr.create(">", self, other)
+
+    def __ge__(self, other: Any) -> "FilterExpr":
+        return FilterExpr.create(">=", self, other)
+
+    def isin(self, other: List[Any]) -> "FilterExpr":
+        return FilterExpr.create("IN", self, other)
+
+    def notin(self, other: List[Any]) -> "FilterExpr":
+        return FilterExpr.create("NIN", self, other)
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}('{self.name}')"
+
+    def to_key(self) -> Key:
+        name = FE_BE_NAME_MAP.get(self.name, self.name)
```

Comment:
This is an internal object -- that's what it's called in the spec.  The actual JS object would look like
```
{section: "summary", name: "potato"}
```

It's called `Key` because that's what the parent calls this object